core/store/src/trie/shard_tries.rs

```diff
@@ -42,12 +42,12 @@ struct ShardTriesInner {
 pub struct ShardTries(Arc<ShardTriesInner>);
 
 impl ShardTries {
-    pub fn new_with_state_snapshot(
+    pub fn new(
         store: Store,
         trie_config: TrieConfig,
         shard_uids: &[ShardUId],
         flat_storage_manager: FlatStorageManager,
-        state_snapshot_config: StateSnapshotConfig,
+        state_snapshot_config: Option<StateSnapshotConfig>,
```

Comment:
Would it make sense to keep it non-optional but do the unwrap_or_default() or StateSnapshotConfig::default() at the call site?